syntax = "proto3";

option cc_enable_arenas = true;

import "tensorflow_serving/apis/input.proto";
import "tensorflow_serving/apis/model.proto";

package tensorflow.serving;

// A single class.
message Class {
  // Label or name of the class.
  string label = 1;
  // Score for this class (e.g., the probability the item belongs to this
  // class).
  float score = 2;
}

// List of classes for a single item
// (tensorflow.Example or tensorflow.InferenceExample.features).
message Classifications {
  repeated Class classes = 1;
}

// For tensorflow.Example this will contain one result.
// For tensorflow.InferenceExample this will contain one result for each
// InferenceExample::features and in the same order as the features.
message ClassificationResult {
  repeated Classifications classifications = 1;
}

// RPC Interfaces

message ClassificationRequest {
  // Model Specification.
  ModelSpec model_spec = 1;

  // Input data.
  tensorflow.serving.Input input = 2;
}

message ClassificationResponse {
  // Result of the classification.
  ClassificationResult result = 1;
}
